tempesta_fw/connection.h

```diff
@@ -105,9 +143,13 @@ tfw_connection_validate_cleanup(TfwConnection *conn)
 	BUG_ON(!conn);
 	BUG_ON(!list_empty(&conn->list));
 	BUG_ON(!list_empty(&conn->msg_queue));
+	BUG_ON(atomic_read(&conn->refcnt) & ~1);
 	BUG_ON(conn->msg);
+
+	spin_lock(&conn->splock);
 	BUG_ON(conn->peer);
 	BUG_ON(conn->sk);
+	spin_unlock(&conn->splock);
```

Comment:
Yeah, agree. Reading a pointer is atomic operation and this is just a validation.